stateMutators.js

```diff
@@ -0,0 +1,111 @@
+const gameState = {
+    basketPos: 50,
+    basketValue: 0,
+    score: 0,
+    livesRemaining: 3,
+    gameLevel: 1,
+    gameMode: 'running',
+    fallingObjects: [],
+}
+
+const setBasketPos = (state, value) => ({
+    ...state,
+    basketPos: value,
+});
+
+const moveBasketLeftKey = (state, value) => ({
+    ...state,
+    basketPos: basketPos - value,
+});
+
+const moveBasketRightKey = (state, value) => ({
+    ...state,
+    basketPos: basketPos + value,
+});
+
+const moveBasketMouse = (state, value) => ({
+    ...state,
+    basketPos: value,
+});
+
+const setBasketValue = (state, value) => ({
+    ...state,
+    basketValue: value,
+});
+
+const resetBasketValue = (state) => ({
+    ...state,
+    basketValue: 0,
+});
+
+const setScore = (state, value) => ({
+    ...state,
+    score: value,
+});
+
+const resetScore = (state) => ({
+    ...state,
+    score: 0,
+});
+
+const setLivesRemaining = (state, value) => ({
+    ...state,
+    livesRemaining: value,
+});
+
+const resetLivesRemaining = (state) => ({
+    ...state,
+    livesRemaining: 3,
+});
+
+const setGameLevel = (state, value) => ({
+    ...state,
+    gameLevel: value,
+});
+
+const resetGameLevel = (state) => ({
+    ...state,
+    gameLevel: 0,
+});
+
+const setGameMode = (state, value) => ({
+    ...state,
+    gameMode: value,
+});
+
+function addFallingObject (state, fallingObj) {
+    state.fallingObjects.push(fallingObj)
+};
+
+function removeFallingObject (state, fallingObj) {
+    let objectIndex = state.fallingObjects.indexOf(fallingObj)
```

Comment:
This is going to cause issues if the `fallingObj` param passed in is not the exact same prototype instance as is stored in state. That could be likely, depending on how the logic is used. Instead it might be easier to keep track of the number of objects spawned and use that count as an ID field in `FallingObject`. Then each object has a unique ID which we can refer to it by, instead of relying on the exact object instance. I would just keep it like this for now, but be aware we might change it soon if it makes sense to